engines/osxnative/sandbox.go

```diff
@@ -136,21 +125,99 @@ func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
 	cmd := exec.Command(s.taskPayload.Command[0], s.taskPayload.Command[1:]...)
 	cmd.Stdout = stdoutLogWriter{s.context}
 	cmd.Stderr = stderrLogWriter{s.context}
+
+	// USER and HOME are treated seperately because their values
+	// depend on either we create the new user successfuly or not
+	processUser := os.Getenv("USER")
+	processHome := os.Getenv("HOME")
+
+	// If we fail to create a new user, the most probable cause is that
+	// we don't have enough permissions. Chances are that we are running
+	// in in a development environment, so do not fail the task to tests
+	// run successfully.
+	u := user{}
+	if err = u.create(); err != nil {
```

Comment:
Thinking about it a little more, i wonder if it would be possible to have the engine configuration option to either use a module for creating the user, or using the user that the engine is running under.  If that config setting says to create a user using some other module, then this .create() stuff can be called and the user returned from that used for the process. Also would allow someone to specify different creation scripts depending on need.  Like if there are some special groups that the user should be part of on one type of worker type but not another.

What do you think?